flink-runtime/src/main/java/org/apache/flink/runtime/shuffle/ShuffleMaster.java

```diff
@@ -84,6 +86,33 @@ CompletableFuture<T> registerPartitionWithProducer(
             PartitionDescriptor partitionDescriptor,
             ProducerDescriptor producerDescriptor);
 
+    /**
+     * Returns all the shuffle descriptors for the partitions in the intermediate data set with the
+     * given id.
+     *
+     * @param intermediateDataSetID The id of the intermediate data set.
+     * @return all the shuffle descriptors for the partitions in the intermediate data set. Null if
+     *     not exist.
+     */
+    default Collection<T> getClusterPartitionShuffleDescriptors(
```

Comment:
What's the use case of this method? Why not get cluster partitions from the partition tracker directly?

Reply:
I mean ClusterPartitionManager and ResourceManagerPartitionTracker, sorry to cause misunderstanding. These classes already have some ability to manage cluster partitions, for example, if a task executor dead, it will remove the lost cluster partitions. But it currently does not support the interface you need which returns a list of ShuffleDescriptors. I think it's better to unify the cluster partition management. BTW, I think it is better to also take care of the case when a cluster partition gets lost in the remote shuffle cluster like what we do currently for the task executor dead.

Reply:
Thanks for the clarification.

If I understand correctly, the `ResourceManagerPartitionTracker` currently only tracks the cluster partition in TaskManager. If we want to unify the cluster partition management, we have to extend the scope of the `ResourceManagerPartitionTracker` so that it also tracks the cluster partitions for remote shuffle service and keeps track of the ShuffleDescriptors of the cluster partitions. The JobMasterClusterPartitionTracker has to register the ShuffleDescriptors of cluster partitions to the ResourceManagerPartitionTracker` when the job is finished. And the ShuffleMaster of the remote shuffle service should be able to report the cluster partition lost in the remote shuffle cluster to ResourceManager.

Please let me know if I understand correctly, then I will go ahead and make the change.